include/pinocchio/algorithm/crba.hxx

```diff
@@ -51,12 +52,70 @@ namespace pinocchio
         else
           data.oMi[i] = data.liMi[i];
 
-        jmodel.jointCols(data.J) = data.oMi[i].act(jdata.S());
+        jmodel.jointJacCols(data.J) = data.oMi[i].act(jdata.S());
 
         data.oYcrb[i] = data.oMi[i].act(model.inertias[i]);
       }
     };
 
+    /// \brief Patch to the crba algorithm for joint mimic (in local convention)
+    template<
+      typename JointModel,
+      typename Scalar,
+      int Options,
+      template<typename, int> class JointCollectionTpl>
+    static inline void mimic_patch_CrbaWorldConventionBackwardStep(
+      const JointModelBase<JointModel> &,
+      const ModelTpl<Scalar, Options, JointCollectionTpl> &,
+      DataTpl<Scalar, Options, JointCollectionTpl> &)
+    {
+    }
+
+    /// \note For each joint the crba computation is done only for the sub cols/rows
+    /// from idx_v to joint.nvSubtree. In the case of the joint j=(i+n) mimicking the joint i,
+    /// the joints i+[1..n-1] will have idx_v greater than the joint j. This patch compute
+    /// this "left out" part of the M matrix.
+    template<typename Scalar, int Options, template<typename, int> class JointCollectionTpl>
+    static inline void mimic_patch_CrbaWorldConventionBackwardStep(
+      const JointModelBase<JointModelMimicTpl<Scalar, Options, JointCollectionTpl>> & jmodel,
+      const ModelTpl<Scalar, Options, JointCollectionTpl> & model,
+      DataTpl<Scalar, Options, JointCollectionTpl> & data)
+    {
+      const JointIndex secondary_id = jmodel.id();
+      const JointIndex primary_id = jmodel.derived().jmodel().id();
+
+      assert(secondary_id > primary_id && "Mimicking joint id is before the primary.");
+
+      size_t ancestor_prim, ancestor_sec;
+      findCommonAncestor(model, primary_id, secondary_id, ancestor_prim, ancestor_sec);
```

Comment:
Instead of computing this data at each call, it's maybe possible to store it in `JointDataMimicTpl`.

Reply:
Yes, it would make sense.

The problem is that both JointModelMimic and JointDataMimic constructors are not "aware" of the full robot model, and don't have access to information on the kinematic tree at all...

Reply:
I agree with you that computing every time seems like a waste, but I am not sure that it is worth storing it.

When `JointDataMimicTpl` is instantiated, it doesn't have any available information regarding the kinematic structure of the whole model. (The `JointModelMimicTpl` does not contain this information either)